scipy/stats/tests/test_hypotests.py

```diff
@@ -202,21 +202,21 @@ def test_input_validation(self):
     def test_auto(self):
         # Test that default method ('auto') chooses intended method
 
-        np.random.seed(1)
+        rng = np.random.RandomState(1)
```

Comment:
Just a note that I tried changing this to `default_rng`; that requires changing `.rand` to `.random` below, and then there's a test failure in this line:
```python
assert auto.pvalue != exact.pvalue
```
That looks like a larger change, best left alone (unclear from a quick glance if the test is not robust, but probably yet since the values do come out the same) - using `RandomState` is hence the simpler thing for this PR.

Reply:
Same for several other tests. These tests are not robust to variations in seeds, e.g. this:
```python
assert np.abs(res1.pvalue - res2.pvalue) > 1e-2
```
is clearly not going to be true when the asymptotic method happens to work well enough.